android/quest/src/test/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModelTest.kt

```diff
@@ -1144,4 +1169,77 @@ class QuestionnaireViewModelTest : RobolectricTest() {
       bundle.entry.any { it.resource is Basic && it.resource.id == "basic-resource-id" },
     )
   }
+
+  @Test
+  fun testSaveExtractedResourcesAddsRelatedEntityLocationMetaTagToExtractedResource() =
+    runBlocking {
+      val bundleSlot = slot<Bundle>()
+      val bundle = bundleSlot.captured
+      val linkId = "linkId"
+      val metaTagId = "UUId123"
+      val questionnaire = extractionQuestionnaire()
+      val questionnaireConfig =
+        questionnaireConfig.copy(
+          resourceIdentifier = "resourceId",
+          resourceType = ResourceType.Location,
+          saveQuestionnaireResponse = false,
+          type = "EDIT",
+          linkIds = listOf(LinkIdConfig(linkId = linkId, LinkIdType.LOCATION)),
+          extractedResourceUniquePropertyExpressions =
+            listOf(
+              ExtractedResourceUniquePropertyExpression(
+                ResourceType.Location,
+                "Observation.code.where(coding.code='obs1').coding.code",
+              ),
+            ),
+        )
+      val previousObs =
+        Observation().apply {
+          id = "previousObs1"
+          code = (CodeableConcept(Coding("http://obsys", "obs1", "Obs 1")))
+        }
+
+      val questionnaireResponse =
+        extractionQuestionnaireResponse().apply {
+          val extractionDate = Date()
+          subject = patient.asReference()
+          val listResource =
+            ListResource().apply {
+              id = metaTagId
+              status = ListResource.ListStatus.CURRENT
+              mode = ListResource.ListMode.WORKING
+              title = CONTAINED_LIST_TITLE
+              date = extractionDate
+            }
+          val listEntryComponent =
+            ListResource.ListEntryComponent().apply {
+              deleted = false
+              date = extractionDate
+              item = previousObs.asReference()
+            }
+          listResource.addEntry(listEntryComponent)
+          addContained(listResource)
+        }
+
+      questionnaireViewModel.saveExtractedResources(
+        bundle = bundle,
+        questionnaire = questionnaire,
+        questionnaireConfig = questionnaireConfig,
+        questionnaireResponse = questionnaireResponse,
+        context = context,
+      )
+
+      val listResource = questionnaireResponse.contained.firstOrNull() as ListResource
+      val resource = listResource.entry.firstOrNull()?.item?.resource
+
+      assertEquals(listResource.id, linkId)
+      Assert.assertNotNull(
+        resource?.meta?.tag?.any { it.code == metaTagId },
+      )
+      Assert.assertNotNull(
+        resource?.meta?.tag?.any {
+          it.system == "https://smartregister.org/related-entity-location-tag-id"
```

Comment:
The `any( `method of the Iterable interface will always return a Boolean, so this test will always pass regardless of the values checked.

Reply:
In my previous comment for the `assertNotNull` was only to replace the if condition that was there previously